src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetGeneratorServiceImpl.java

```diff
@@ -42,8 +43,8 @@ public Dataset generate(String id, String name, Country country, Language langua
             .language(language)
             .content(record)
             .build())
-        .collect(Collectors.toList());
+        .collect(toSet());
 
-    return new Dataset(id, recordObjectList);
+    return new Dataset(id, recordObjectList, totalRecords - recordObjectList.size());
```

Comment:
I don't think this is the right way to count duplicates. I think we should be looking at the record IDs alone (and the equals method in Record look at other properties as well). Perhaps two records with the same ID but different languages exist, but they should still be counted as duplicates I think.

Reply:
Yes I will update the equals method for the Record class